snapcraft/models/project.py

```diff
@@ -615,41 +611,40 @@ class Project(models.Project):
     compression: Literal["lzo", "xz"] = "xz"
-    version: Optional[VersionStr]  # type: ignore[assignment]
-    donation: Optional[UniqueStrList]
+    version: VersionStr | None = None
+    donation: UniqueStrList | None = None
     # snapcraft's `source_code` is more general than craft-application
-    source_code: Optional[UniqueStrList]  # type: ignore[assignment]
-    contact: Optional[UniqueStrList]  # type: ignore[assignment]
-    issues: Optional[UniqueStrList]  # type: ignore[assignment]
-    website: Optional[UniqueStrList]
-    type: Optional[Literal["app", "base", "gadget", "kernel", "snapd"]]
-    icon: Optional[str]
+    source_code: UniqueStrList | None = None  # type: ignore[assignment]
+    contact: UniqueStrList | None = None  # type: ignore[assignment]
+    issues: UniqueStrList | None = None  # type: ignore[assignment]
+    website: UniqueStrList | None = None
+    type: Literal["app", "base", "gadget", "kernel", "snapd"] | None = None
+    icon: str | None = None
     confinement: Literal["classic", "devmode", "strict"]
-    layout: Optional[
-        Dict[str, Dict[Literal["symlink", "bind", "bind-file", "type"], str]]
-    ]
-    grade: Optional[Literal["stable", "devel"]]
-    architectures: List[Union[str, Architecture]] | None = None
+    layout: (
+        dict[str, dict[Literal["symlink", "bind", "bind-file", "type"], str]] | None
+    ) = None
+    grade: Literal["stable", "devel"] | None = None
+    architectures: list[str | Architecture] | None = None
     platforms: dict[str, Platform] | None = None  # type: ignore[assignment,reportIncompatibleVariableOverride]
     assumes: UniqueStrList = cast(UniqueStrList, [])
-    package_repositories: Optional[List[Dict[str, Any]]]
-    hooks: Optional[Dict[str, Hook]]
-    passthrough: Optional[Dict[str, Any]]
-    apps: Optional[Dict[str, App]]
-    plugs: Optional[Dict[str, Union[ContentPlug, Any]]]
-    slots: Optional[Dict[str, Any]]
-    lint: Optional[Lint]
-    epoch: Optional[str]
-    adopt_info: Optional[str]
-    system_usernames: Optional[Dict[str, Any]]
-    environment: Optional[Dict[str, Optional[str]]]
-    build_packages: Optional[GrammarStrList]
-    build_snaps: Optional[GrammarStrList]
-    ua_services: Optional[UniqueStrList]
-    provenance: Optional[str]
-    components: Optional[Dict[ProjectName, Component]]
+    hooks: dict[str, Hook] | None = None
+    passthrough: dict[str, Any] | None = None
+    apps: dict[str, App] | None = None
+    plugs: dict[str, ContentPlug | Any] | None = None
+    slots: dict[str, Any] | None = None
+    lint: Lint | None = None
+    epoch: str | None = None
+    adopt_info: str | None = None
+    system_usernames: dict[str, Any] | None = None
+    environment: dict[str, str | None] | None = None
+    build_packages: Grammar[list[str]] | None = None
+    build_snaps: Grammar[list[str]] | None = None
+    ua_services: UniqueStrList | None = None
```

Comment:
```suggestion
    ua_services: UniqueList[str] | None = None
```

or possibly a `set` here too